diff --git a/configure.ac b/configure.ac
index e9f3adf..7dcac7b 100644
--- a/configure.ac
+++ b/configure.ac
@@ -46,7 +46,7 @@ dnl **************************************************************************
 dnl Add warning flags
 dnl **************************************************************************
 
-AX_COMPILER_FLAGS([WARN_CFLAGS], [WARN_LDFLAGS])
+#AX_COMPILER_FLAGS([WARN_CFLAGS], [WARN_LDFLAGS])
 
 dnl **************************************************************************
 dnl Initialize automake
