diff --git a/modules/notification-area/na-applet.c b/modules/notification-area/na-applet.c
index f866bbe..3412b11 100644
--- a/modules/notification-area/na-applet.c
+++ b/modules/notification-area/na-applet.c
@@ -148,7 +148,7 @@ na_applet_init (NaApplet *na)
   AtkObject *atk_object;
   GtkOrientation orientation;
 
-  flags = GP_APPLET_FLAGS_HAS_HANDLE | GP_APPLET_FLAGS_EXPAND_MINOR;
+  flags = GP_APPLET_FLAGS_EXPAND_MINOR;
   gp_applet_set_flags (GP_APPLET (na), flags);
 
   atk_object = gtk_widget_get_accessible (GTK_WIDGET (na));
diff --git a/modules/notification-area/na-tray.c b/modules/notification-area/na-tray.c
index f9ee219..a2aec95 100644
--- a/modules/notification-area/na-tray.c
+++ b/modules/notification-area/na-tray.c
@@ -26,7 +26,7 @@
 #include "na-tray-manager.h"
 #include "fixedtip.h"
 
-#define ICON_SPACING 1
+#define ICON_SPACING 0
 #define MIN_BOX_SIZE 3
 
 struct _NaTray
