diff --git a/gnome-panel/panel-multiscreen.c b/gnome-panel/panel-multiscreen.c
index 756fba1..dcd4b5c 100644
--- a/gnome-panel/panel-multiscreen.c
+++ b/gnome-panel/panel-multiscreen.c
@@ -514,7 +514,7 @@ panel_multiscreen_x (GdkScreen *screen,
 	n_screen = gdk_screen_get_number (screen);
 
 	g_return_val_if_fail (n_screen >= 0 && n_screen < screens, 0);
-	g_return_val_if_fail (monitor >= 0 && monitor < monitors [n_screen], 0);
+	g_return_val_if_fail (monitor >= 0, 0);
 
 	return geometries [n_screen][monitor].x;
 }
@@ -528,7 +528,7 @@ panel_multiscreen_y (GdkScreen *screen,
 	n_screen = gdk_screen_get_number (screen);
 
 	g_return_val_if_fail (n_screen >= 0 && n_screen < screens, 0);
-	g_return_val_if_fail (monitor >= 0 && monitor < monitors [n_screen], 0);
+	g_return_val_if_fail (monitor >= 0, 0);
 
 	return geometries [n_screen][monitor].y;
 }
@@ -542,7 +542,7 @@ panel_multiscreen_width (GdkScreen *screen,
 	n_screen = gdk_screen_get_number (screen);
 
 	g_return_val_if_fail (n_screen >= 0 && n_screen < screens, 0);
-	g_return_val_if_fail (monitor >= 0 && monitor < monitors [n_screen], 0);
+	g_return_val_if_fail (monitor >= 0, 0);
 
 	return geometries [n_screen][monitor].width;
 }
@@ -556,7 +556,7 @@ panel_multiscreen_height (GdkScreen *screen,
 	n_screen = gdk_screen_get_number (screen);
 
 	g_return_val_if_fail (n_screen >= 0 && n_screen < screens, 0);
-	g_return_val_if_fail (monitor >= 0 && monitor < monitors [n_screen], 0);
+	g_return_val_if_fail (monitor >= 0, 0);
 
 	return geometries [n_screen][monitor].height;
 }
